stake BYX + initial distribution

YvesToiser opened the pull request:
Basic Staking of BYX (no rewards)
Basic functions in all contracts (receive, fallback, withdraw)
Initial token distribution system

I have not tested this code yet. I will do a test suite this afternoon and bring corrections if needed.

0xB19 commented on truffle/contracts/StakingManager.sol:
```diff
@@ -0,0 +1,75 @@
+     */
+    function withdrawStake(uint _amount) external {
+        require(BYXStaked[msg.sender] >= _amount, "Not enough BYX staked");
+        totalBYXStaked += _amount;
```
plutot -= non ?

0xB19 commented on truffle/contracts/StakingManager.sol:
```diff
@@ -0,0 +1,75 @@
+     * @param _amount the amount to deposit in staking.
+     */
+    function depositStake(uint _amount) external {
+        require(BYX.balanceOf(msg.sender) >= _amount, "Not enough BYX staked");
```
Le message d'erreur m'a un peut perturbé pour la compréhension de la ligne. J'aurais mis quelque chose comme "Not enough BYX in address balance"

0xB19 commented on truffle/contracts/StakingManager.sol:
```diff
@@ -0,0 +1,75 @@
+    function withdrawStake(uint _amount) external {
+        require(BYXStaked[msg.sender] >= _amount, "Not enough BYX staked");
+        totalBYXStaked += _amount;
+        BYXStaked[msg.sender] += _amount;
```
Meme chose, -=

0xB19 commented on truffle/contracts/StakingManager.sol:
```diff
@@ -0,0 +1,75 @@
+     *
+     * @param _amount the amount to deposit in staking.
+     */
+    function depositStake(uint _amount) external {
```
On peut verifier que le montant ne soit pas 0

YvesToiser commented:
Fixed
I also added a getter for the user staked amount (that will be needed by frontend
I switched the functions between an external that call the internal for gas optimisation (like openzeppellin always do)
